packages/invoices/src/InboundShipment/api/hooks.ts

```diff
@@ -125,17 +125,23 @@ export const useInboundLines = (itemId?: string) => {
 };
 
 export const useInboundItems = () => {
-  const { sortBy, onChangeSortBy } = useSortBy<InboundItem>({
-    key: 'itemName',
-  });
-
-  const selectItems = useCallback((invoice: InboundFragment) => {
-    return inboundLinesToSummaryItems(
+  const { sort } = useQueryParamsStore();
+  const { sortBy, onChangeSortBy } = sort;
+  const selectItems = (invoice: InboundFragment) =>
+    inboundLinesToSummaryItems(
       invoice.lines.nodes.filter(line => isA.stockInLine(line))
-    ).sort(
-      SortUtils.getDataSorter(sortBy.key as keyof InboundItem, !!sortBy.isDesc)
-    );
-  }, []);
+    )
+      .map(item => ({
```

Comment:
a neat little bug here: the hook was using the rows, mapping them to `SummaryItems` which do not have the properties which are sorted by. 

So the `getDataSorter` is unable to sort the rows, which means that the `OK & Next` button was using the rows in the wrong order. I've popped in a prop, of the sort key, which allows correct sorting.